stacks/spica/src/passport/pages/identify/identify.component.ts

```diff
@@ -15,13 +16,26 @@ export class IdentifyComponent implements OnInit {
 
   strategies: Observable<Strategy[]>;
 
-  constructor(public passport: PassportService, public router: Router) {}
+  constructor(
+    public passport: PassportService,
+    public router: Router,
+    private activatedRoute: ActivatedRoute
+  ) {}
 
   ngOnInit() {
     this.strategies = this.passport.getStrategies();
-    if (this.passport.identified) {
-      this.router.navigate([""]);
-    }
+    this.activatedRoute.queryParams
+      .pipe(
+        map(params => params.token),
+        tap(token => {
+          if (token) this.passport.token = token;
+          if (this.passport.identified) {
+            this.router.navigate([""]);
+          }
+        }),
+        take(1)
+      )
+      .subscribe();
```

Comment:
```suggestion
    this.activatedRoute.queryParams
      .pipe(
        filter(params => params.token)
        take(1)
      )
      .subscribe(params => {
           if (params.token) {
              this.passport.token = token;
           }
          if (this.passport.identified) {
            this.router.navigate([""]);
          }
      });
```
I think this one more appropriate for this logic.